includes/Modules/Analytics_4.php

```diff
@@ -805,19 +805,25 @@ protected function create_data_request( Data_Request $data ) {
 					);
 				}
 
-				// If the scope is not set or is not `EVENT`, set it to `EVENT`.
-				if ( empty( $custom_dimention_data['scope'] ) || 'EVENT' !== $custom_dimention_data['scope'] ) {
-					$custom_dimention_data['scope'] = 'EVENT';
+				// Validate against the `DimensionScope` enum.
+				$valid_scopes = array( 'EVENT', 'USER', 'ITEM' );
+
+				if ( ! isset( $custom_dimension_data['scope'] ) || ! in_array( $custom_dimension_data['scope'], $valid_scopes, true ) ) {
+					return new WP_Error(
+						'invalid_scope',
+						/* translators: %s: Invalid scope */
+						sprintf( __( 'Invalid scope: %s.', 'google-site-kit' ), $custom_dimension_data['scope'] ),
```

Comment:
Referencing a non-existent array index will result in an `Undefined index` run-time notice being reported, which although not fatal we do try to avoid.

To avoid this, I think we should handle the `! isset()` and `! in_array()` cases separately. Actually, maybe I wasn't as clear as I could have been in my previous comment; seeing as we do only expect to use `EVENT` scoped custom dimensions for the foreseeable future, I _do_ think it would be reasonable to default to `EVENT` in the case where the scope isn't provided at all, i.e. the `! isset()` case. It's just when the scope _is_ provided that I think we should validate it against the enum values, to avoid giving a false positive validation for an invalid value.

So, for the sake of a clear direction here, could you please update this so if the scope field is not set at all, it's defaulted to `EVENT`, but if it is provided, it's validated against the enum values?

Reply:
Updated ✅